apps/native/app/src/stores/auth-store.ts

```diff
@@ -59,134 +61,143 @@ const getAppAuthConfig = () => {
-  },
-  async logout() {
-    // Clear all MMKV storages
-    clearAllStorages()
-
-    // Clear push token if exists
-    const pushToken = notificationsStore.getState().pushToken
-    if (pushToken) {
-      notificationsStore.getState().deletePushToken(pushToken)
-    }
-    notificationsStore.getState().reset()
-
-    const appAuthConfig = getAppAuthConfig()
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const tokenToRevoke = get().authorizeResult!.accessToken!
-    try {
-      await revoke(appAuthConfig, {
-        tokenToRevoke,
-        includeBasicAuth: true,
-        sendClientId: true,
-      })
-    } catch (e) {
-      // NOOP
-    }
-
-    const client = await getApolloClientAsync()
-    await client.cache.reset()
-    await Keychain.resetGenericPassword({ service: KEYCHAIN_AUTH_KEY })
-    set(
-      (state) => ({
-        ...state,
-        authorizeResult: undefined,
-        userInfo: undefined,
-      }),
-      true,
-    )
-    return true
-  },
-}))
+      async login() {
+        const appAuthConfig = getAppAuthConfig()
+        const authorizeResult = await authorize({
+          ...appAuthConfig,
+          additionalParameters: {
+            prompt: 'login',
+            prompt_delegations: 'true',
+            ui_locales: preferencesStore.getState().locale,
+            externalUserAgent: 'yes',
+          },
+        })
+
+        if (authorizeResult) {
+          await Keychain.setGenericPassword(
+            KEYCHAIN_AUTH_KEY,
+            JSON.stringify(authorizeResult),
+            { service: KEYCHAIN_AUTH_KEY },
+          )
+          set({ authorizeResult })
+          return true
+        }
+        return false
+      },
+      async logout() {
+        // Clear all MMKV storages
+        clearAllStorages()
+
+        // Clear push token if exists
+        const pushToken = notificationsStore.getState().pushToken
+        if (pushToken) {
+          notificationsStore.getState().deletePushToken(pushToken)
+        }
+        notificationsStore.getState().reset()
+
+        const appAuthConfig = getAppAuthConfig()
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const tokenToRevoke = get().authorizeResult!.accessToken!
+        try {
+          await revoke(appAuthConfig, {
+            tokenToRevoke,
+            includeBasicAuth: true,
+            sendClientId: true,
+          })
+        } catch (e) {
+          // NOOP
+        }
+
+        const client = await getApolloClientAsync()
+        await client.cache.reset()
+        await Keychain.resetGenericPassword({ service: KEYCHAIN_AUTH_KEY })
+        set(
+          (state) => ({
+            ...state,
+            authorizeResult: undefined,
+            userInfo: undefined,
+          }),
+          true,
+        )
+        return true
+      },
+    }),
+    {
+      name: 'auth_01',
+      getStorage: () => AsyncStorage,
+      whitelist: ['userInfo'],
+    },
```

Comment:
Only thing I changed is to add `persist(` at the top and then this object with three lines at the bottom 😊